### PR TITLE
WebRequestPsCmdlet: handle link relation keys case-insensitively

### LINKED ISSUE #6146: RelationLink dictionary keys should be case-insensitive
Per RFC 8288, relation types on `Link` response headers MUST be compared in a case-insensitive fashion. HTTP servers SHOULD send relation types as all lower case, but parsing them MUST be case-insensitive. 

Currently, the `RelationLink` dictionary we create is case sensitive. This should be change to case-insensitive, An HTTP server sending more than one of the same relation type (case-insensitive) is not RFC compliant, therefore a case-insensitive dictionary should not encounter keys of the same characters with different casing.

https://github.com/PowerShell/PowerShell/blob/0ad9fc9d7057bbe50c77b7d297f102a68380918b/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs#L1593

Steps to reproduce
------------------

```powershell
$uri = 'https://api.github.com/repositories/49609581/issues?page=2'
$response = Invoke-WebRequest -Uri $Uri
$response.RelationLink.next -eq 'https://api.github.com/repositories/49609581/issues?page=3'
$response.RelationLink.Next -eq 'https://api.github.com/repositories/49609581/issues?page=3'
```

Expected behavior
-----------------

```none
True
True
```

Actual behavior
---------------

```none
True
False
```

Environment data
----------------

```none
Name                           Value
----                           -----
PSVersion                      6.0.1
PSEdition                      Core
GitCommitId                    v6.0.1
OS                             Microsoft Windows 10.0.16299
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

### PR DESCRIPTION
**Pull request by rjmholt**
## PR Summary
* Fix #6146, where web request cmdlets use case-sensitive keys for link relations, contrary to [RFC 8288](https://tools.ietf.org/html/rfc8288).
* Add tests to monitor this property.

All credit goes to @markekraus.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by markekraus** [comment]
Hi! Thank you for your contribution. Please push a commit with `[Feature]` in the commit message to ensure the Feature tests are run.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1 [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿# Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
-# This is a Pester test suite which validate the Web cmdlets.
+# This is a Pester test suite which validates the Web cmdlets.
```
Comment (markekraus) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1]: Please revert this change. Since it is unrelated to your code change, it should be corrected in a separate PR.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1 [hunk 2]
```diff
@@ -733,6 +733,19 @@ Describe "Invoke-WebRequest tests" -Tags "Feature" {
         $result.Output.RelationLink["self"] | Should -BeExactly "${baseUri}?maxlinks=5&linknumber=2&type=multiple"
     }
 
+
+    It "Validate Invoke-WebRequest RelationLink keys are treated case-insensitively" {
+        $uri = Get-WebListenerUrl -Test 'Link' -Query @{maxlinks = 5; linknumber = 2; type = 'multiple'}
+        $command = "Invoke-WebRequest -Uri '$uri'"
+        $result = ExecuteWebCommand -command $command
+
+        $result.Output.RelationLink["next"] -eq $result.Output.RelationLink["Next"] | Should -BeTrue
```
Comment (markekraus) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1]: Please redo these tests to be like the following:

```powershell
$result.Output.RelationLink["next"] | Should -Not -BeNullOrEmpty
$result.Output.RelationLink["next"] | Should -BeExactly $result.Output.RelationLink["Next"]
```

Repeat that pattern for the other assertions in this block.

This way we ensure that we are not somehow comparing `$null` to `$null` and getting a false pass.

**Review by markekraus — CHANGES_REQUESTED** [review]
Please adjust the tests and revert the unrelated change. Also, please include `[feature]` in your commit messages for this PR.

**Review by markekraus — APPROVED** [review]

**Review by daxian-dbw — APPROVED** [review]
